mlflow/spark.py

```diff
@@ -50,13 +51,21 @@
 def get_default_conda_env():
     """
     :return: The default Conda environment for MLflow Models produced by calls to
-             :func:`save_model()` and :func:`log_model()`.
+             :func:`save_model()` and :func:`log_model()`. This Conda environment
+             contains the current version of PySpark that is installed on the caller's
+             system. ``dev`` versions of PySpark are replaced with stable versions in
+             the resulting Conda environment (e.g., if you are running PySpark version
+             ``2.4.5.dev0``, invoking this method produces a Conda environment with a
+             dependency on PySpark version ``2.4.5``).
     """
     import pyspark
+    # Strip the suffix from `dev` versions of PySpark, which are not
+    # available for installation from Anaconda or PyPI
+    pyspark_version = re.sub(r"dev.*", "", pyspark.__version__).rstrip(".")
```

Comment:
Why not just sub the expression `"\.dev.*" with the empty string? It seems weird to both sub and rstrip.

Reply:
I'd like to handle the case where `dev` isn't prefixed with a dot, which seems like something that could reasonably occur. I'm more than happy to move this into the regular expression.